src/borrower/feed/navfeed.sol

```diff
@@ -273,106 +161,272 @@ contract NAVFeed is BaseNFTFeed, Interest, Buckets, FixedPoint {
+
+    // TODO: we could also remove the writeOffGroupIndex_ argument, loop over the
+    // writeOffGroups and find the latest writeoff group which can be applied.
+    function writeOff(uint loan, uint writeOffGroupIndex_) public {
+        require(!writeOffOverride[loan], "already-overridden");
+
+        bytes32 nftID_ = nftID(loan);
+        WriteOffGroup memory writeOffGroup_ = writeOffGroups[writeOffGroupIndex_];
+        uint maturityDate_ = maturityDate[nftID_];
+        require(block.timestamp >= maturityDate_ + writeOffGroup_.overdueDays * 60 * 60 * 24, "too-early");
+
+        uint currentRate = pile.loanRates(loan);
+        require(currentRate < WRITEOFF_RATE_GROUP_START || writeOffGroup_.overdueDays > writeOffGroups[currentRate].overdueDays, "cannot-go-back");
+
+        _writeOff(loan, writeOffGroupIndex_, nftID_, maturityDate_);
     }
 
-    function calcDiscount(uint amount, uint normalizedBlockTimestamp, uint maturityDate_) public view returns (uint result) {
-        return rdiv(amount, rpow(discountRate.value, safeSub(maturityDate_, normalizedBlockTimestamp), ONE));
+    function overrideWriteOff(uint loan, uint writeOffGroupIndex_) public auth {
+        writeOffOverride[loan] = true;
+        
+        bytes32 nftID_ = nftID(loan);
+        uint maturityDate_ = maturityDate[nftID_];
+        _writeOff(loan, writeOffGroupIndex_, nftID_, maturityDate_);
     }
 
-    function calcTotalDiscount() public view returns(uint) {
-        (uint sum, ) = _calcTotalDiscount();
-        return sum;
+    function _writeOff(uint loan, uint writeOffGroupIndex_, bytes32 nftID_, uint maturityDate_) internal {
+        if (pile.loanRates(loan) < WRITEOFF_RATE_GROUP_START) {
+            // this is the first writeoff, remove future value for loan from bucket
+            uint fv = futureValue[nftID_];
+            buckets[maturityDate_] = safeSub(buckets[maturityDate_], fv);
+
+            if (uniqueDayTimestamp(lastNAVUpdate) >= maturityDate_) {
+                overdueButNotWrittenOff = secureSub(overdueButNotWrittenOff, fv);
+            }
+
+            latestDiscount = secureSub(latestDiscount, fv);
+            latestNAV = secureSub(latestNAV, fv);
+        }
+
+        pile.changeRate(loan, WRITEOFF_RATE_GROUP_START + writeOffGroupIndex_);
     }
 
-    // calculates the total discount of all buckets with a timestamp > block.timestamp
-    function _calcTotalDiscount() internal view returns(uint, uint) {
-        uint normalizedBlockTimestamp = uniqueDayTimestamp(block.timestamp);
-        uint sum = 0;
+    // --- NAV calculation ---
+    function currentNAV() public view returns(uint) {
+        uint nnow = uniqueDayTimestamp(block.timestamp);
+        uint nLastUpdate = uniqueDayTimestamp(lastNAVUpdate);
+        uint overdueButNotWrittenOff_ = 0;
+        for (uint i = nLastUpdate; i < nnow; i = i + 1 days) {
+            overdueButNotWrittenOff_ = safeAdd(overdueButNotWrittenOff_, buckets[i]);
+        }
```

Comment:
It would be nice to calculate the `overdueButNotWrittenOff_` in the `currentDiscount` loop.
I think this would be cheaper because the additional `i` increment would not be required

Reply:
Maybe we can rename the  totalDiscount function into presentValue and it returns the totalDiscount and an up to date `overdueButNotWrittenOff` or the totalDiscount + overdueButNotWrittenOff